### ISSUE TITLE
can't load videos on WeChat when Brook is connected lol 

### ISSUE THREAD

**Issue by godbout** [txthinking/brook#1337]
happens at least on iPhone. once i'm connected through Brook, some of the traffic doesn't go through. like can't load WeChat videos. some sites return a 502 bad gateway. and even localhosts on my Mac don't load lol. any info i can give to debug? (if you care of course)

**Comment by txthinking** [comment]
>  like can't load WeChat videos.

Yes I know this issue, I have tried debug it, but not success, maybe wechat want to p2p or udp packet size, sorry I have no idea with wechat.

> some sites return a 502 bad gateway

This is usually not a problem for Brook. maybe related service your want to visit and your IP, because 502 is not an error at the transport layer(TCP/UDP). it is a application layer(HTTP) error

> localhosts on my Mac don't load

because Brook considers `localhost` to be a domain name, no different with `google.com`.
But you listen and visit [::1]:8080, or force localhost to loopback one of v4 ip or v6 ip and block one, or both with [module](https://github.com/txthinking/brook/blob/master/programmable/modules/hosts.tengo)

**Comment by godbout** [comment]
> Yes I know this issue, I have tried debug it, but not success, maybe wechat want to p2p or udp packet size, sorry I have no idea with wechat.

alright. thank you Sir!

> This is usually not a problem for Brook. maybe related service your want to visit and your IP, because 502 is not an error at the transport layer(TCP/UDP). it is a application layer(HTTP) error

makes sense. the first URL that comes to mind is: https://appleid.apple.com
both failing on iPhone and Mac, but yeah, might be related to Apple's services.

> because Brook considers `localhost` to be a domain name, no different with `google.com`. But you listen and visit [::1]:8080, or force localhost to loopback one of v4 ip or v6 ip and block one, or both with [module](https://github.com/txthinking/brook/blob/master/programmable/modules/hosts.tengo)

interesting. 谢谢你。that answers. can close once you read 👍

**Comment by godbout** [comment]
clear 👍️ 辛苦了

**Comment by godbout** [comment]
a little more strange thing regarding the "localhost". i have local dev sites running at urls like "https://hq.sleeplessmind.com.mo.test/" for example. loading all fine on Safari, but keep loading and can't never connect using Brave 😅️ as soon as i close Brook, then Brave load the dev site all fine 🤔️ tried to go through the settings but can't find anything. any idea? 🥺️ 辛苦了

**Comment by godbout** [comment]
> https://www.txthinking.com/talks/articles/brook-fakedns-en.article https://www.txthinking.com/talks/articles/brook-ipv6-en.article it looks like Brave has own DOH? by [unknown](https://t.me/txthinking)

omg i'm learning so much 哈哈哈哈哈哈

had no idea about DOH. my god Brook is such a genius (means you). so the way i've fixed this on my side is playing with the following Settings in Brave:

<img width="716" alt="Screenshot 2024-02-29 at 13 40 32" src="https://github.com/txthinking/brook/assets/121373/eabb0a4b-90cd-4cdf-b262-46f860e34cc1">

enable/disable the secure DNS didn't change anything. then i've tried different DNS and they all failed UNTIL i came back to the OS one. then it started working lol. so maybe an issue with Brave not refreshing when Brooks connects? no idea. anyways. solved now. glad i've learned a bit and understand a bit more. 谢谢你

**Comment by godbout** [comment]
today i was in 中国 and i kept having issues with running web proxies in a VM, especially using shared network. and especially Zap (seems Burp was ok) and i think this is related to the same issue, with the local stuff not working. just adding here because of my own Alzheimer. and also i see that i had somehow find some solutions for Brave and post it right above but had forgotten about it lol

**Comment by txthinking** [comment]
maybe wechat use p2p to lower costs but hurts you, not sure

**Comment by godbout** [comment]
no worries. 辛苦了
